sdk/gossip/client/client.go

```diff
@@ -347,9 +348,18 @@ func (c *Client) SendWithoutWait(ctx context.Context, abr *services.AddBlockRequ
 }
 
 func (c *Client) SubscribeToRounds(ctx context.Context, ch chan *types.RoundConfirmationWrapper) (subscription, error) {
-	return c.subscriber.stream.Subscribe(func(evt interface{}) {
-		ch <- evt.(*ValidationNotification).RoundConfirmation
-	}), nil
+	doneCh := ctx.Done()
+	var sub subscription
+	sub = c.subscriber.stream.Subscribe(func(evt interface{}) {
+		select {
+		case ch <- evt.(*ValidationNotification).RoundConfirmation:
+		case <-doneCh:
```

Comment:
I added done handling here and also I allow the channel publish to *fail* instead of hang because an EventSubscriber hangs on each function it's calling.